patches/minecraft/net/minecraft/tags/TagLoader.java.patch

```diff
@@ -1,11 +1,108 @@
 --- a/net/minecraft/tags/TagLoader.java
 +++ b/net/minecraft/tags/TagLoader.java
-@@ -164,7 +_,7 @@
+@@ -66,7 +_,12 @@
+ 
+                   String s1 = resource.m_215506_();
+                   tagfile.f_215959_().forEach((p_215997_) -> {
+-                     list.add(new TagLoader.EntryWithSource(p_215997_, s1));
++                     // FORGE: Specify this tag is being added
++                     list.add(new TagLoader.EntryWithSource(p_215997_, s1, true));
++                  });
++                  // FORGE: Add removed tags
++                  tagfile.removeEntries().forEach(entry1 -> {
++                     list.add(new TagLoader.EntryWithSource(entry1, s1, false));
+                   });
+                } catch (Throwable throwable1) {
+                   if (reader != null) {
+@@ -119,21 +_,30 @@
+ 
+    }
+ 
++   /** @deprecated Forge: Use {{@link #buildPair(net.minecraft.tags.TagEntry.Lookup, List)} instead. */
++   @Deprecated
+    private Either<Collection<TagLoader.EntryWithSource>, Collection<T>> m_215978_(TagEntry.Lookup<T> p_215979_, List<TagLoader.EntryWithSource> p_215980_) {
+-      ImmutableSet.Builder<T> builder = ImmutableSet.builder();
++      return buildSets(p_215979_, p_215980_).mapRight(p -> p.additive());
++   }
++
++   // FORGE: Build both the additive and subtractive sets for the given tags
++   private Either<Collection<TagLoader.EntryWithSource>, AdditiveSubtractiveSets<T>> buildSets(TagEntry.Lookup<T> lookup, List<TagLoader.EntryWithSource> entries) {
```

Comment:
This could have an `EXC` entry added.

Reply:
What should the EXC entry be in this case?